app/controllers/ht_approval_requests_controller.rb

```diff
@@ -26,8 +31,15 @@ def update # rubocop:disable Metrics/MethodLength
     redirect_to action: 'show'
   end
 
-  def all_sent?
-    @reqs.all? { |r| r.sent.present? }
+  def status_counts
+    counts = {unsent: 0, sent: 0, expired: 0, received: 0}
+    @reqs.each do |r|
+      counts[:unsent] += 1 if r.sent.nil?
+      counts[:sent] += 1 if r.sent.present?
+      counts[:expired] += 1 if r.expired?
+      counts[:received] += 1 if r.received.present?
+    end
+    counts
```

Comment:
Maybe it would be better to just rename the DB column `received` -> `approved`. For the UI "Approved" is clearer. "Received" was initially chosen for the "sent/received" duo.

In any case, I'll adopt your more elegant second solution and work around the fact that some states won't have an entry, rather than having a zero count.